Delete git-clone basic-auth secret when deleting a pipelinerun

chmouel opened the pull request:
When deleting a pipelinerun, the git-clone basic-auth secret should be deleted
as well, but in some cases of conflicts and the ownerRef not being set, the
secret is not deleted and we need to delete it manually.

Fixes #1297

chmouel committed:
Delete git-clone basic-auth secret when deleting a pipelinerun

When deleting a pipelinerun, the git-clone basic-auth secret should be deleted
as well, but in some cases of conflicts and the ownerRef not being set, the
secret is not deleted and we need to delete it manually.

Fixes #1297

sm43 commented on pkg/kubeinteraction/cleanups.go:
```diff
@@ -45,6 +43,12 @@ func (k Interaction) CleanupPipelines(ctx context.Context, logger *zap.SugaredLo
+			// Try to Delete the secret created for git-clone basic-auth, it should have been created with a owneref on the pipelinerun and due being deleted when the pipelinerun is deleted
+			// but in some cases of conflicts and the ownerRef not being set, the secret is not deleted and we need to delete it manually.
+			if secretName, ok := pr.GetAnnotations()[keys.GitAuthSecret]; ok {
+				_ = k.Run.Clients.Kube.CoreV1().Secrets(repo.GetNamespace()).Delete(ctx, secretName, metav1.DeleteOptions{})
```
we do have `kubeinteraction` pkg dealing with secret create and update
may be we wanna move this there??!!

chmouel committed:
Add test for deleting secret with PipelineRun

and some refactoring along the way. I separated the commits to make it
easier to review from the implementation.

piyush-garg commented:
/lgtm
